File: frontend/src/types/game.ts
export type GameMode = 'quick' | 'full';
export type GamePhase =
  | 'motivation'
  | 'precheck'
  | 'track_select'
  | 'preparation'
  | 'opening'
  | 'stable'
  | 'crisis'
  | 'report';

export interface GameMetrics {
  cash: number;
  debt: number;
  dailyNetCashFlow: number;
  weeklyNetCashFlow: number;
  survivalDays: number;
  familyPressure: number;
  energy: number;
  reputation: number;
  repeatRate: number;
  information: number;
  impulse: number;
}

export interface Role {
  id: string;
  name: string;
  description: string;
  initialMetrics: GameMetrics;
  advantages: string[];
  risks: string[];
}

export interface BusinessTrack {
  id: string;
  name: string;
  description: string;
  baseParams: Record<string, number>;
  strengths: string[];
  risks: string[];
  uniqueSystems: string[];
}

export interface OpeningPackage {
  id: string;
  name: string;
  description: string;
  effect: Record<string, number>;
}

export interface BootstrapContent {
  contentVersion: string;
  roles: Role[];
  businessTracks: BusinessTrack[];
  openingPackages: OpeningPackage[];
  phases: Array<{ id: GamePhase; name: string }>;
}

export interface GameSession {
  id: string;
  contentVersion: string;
  mode: GameMode;
  status: 'active' | 'finished' | 'abandoned';
  phase: GamePhase;
  day: number;
  week: number;
  roleId: string;
  businessTrackId?: string;
  openingPackageId?: string;
  metrics: GameMetrics;
  flags: Record<string, boolean | number | string>;
}

export interface EventCard {
  id: string;
  phase: GamePhase;
  category: string;
  title: string;
  content: string;
  effect: Record<string, number>;
  weight: number;
}

export interface DecisionCard {
  id: string;
  title: string;
  phase: GamePhase;
  description: string;
  cost: Record<string, number>;
  effect: Record<string, number>;
  risk: string[];
  insight: string;
}

export interface AvailableDecisionsResponse {
  phase: GamePhase;
  day: number;
  week: number;
  event?: EventCard;
  decisions: DecisionCard[];
}

export interface LedgerEntry {
  periodType: 'day' | 'week';
  periodIndex: number;
  revenueAmount: number;
  materialCost: number;
  laborCost: number;
  rentCost: number;
  platformFee: number;
  discountCost: number;
  marketingCost: number;
  wasteCost: number;
  utilityCost: number;
  repairCost: number;
  operationNetCashFlow: number;
  familyCost: number;
  loanPayment: number;
  familyNetCashFlow: number;
  detail: Record<string, number | string | string[]>;
}

export interface RiskAlert {
  level: 'info' | 'warning' | 'danger';
  message: string;
}

export interface SimulateResponse {
  session: GameSession;
  eventResult: {
    title: string;
    message: string;
  };
  actionLogs: string[];
  ledger: LedgerEntry;
  riskAlerts: RiskAlert[];
}

export interface Report {
  endingType: string;
  title: string;
  summary: string;
  totalRevenue: number;
  totalNetCashFlow: number;
  maxLossSources: string[];
  rightDecisions: string[];
  dangerousDecisions: string[];
  suggestion: string;
  shareText: string;
  reportDetail: Record<string, number | string | string[]>;
}

File: backend/app/schemas/game.py
from typing import Literal

from pydantic import BaseModel, Field

GameMode = Literal["quick", "full"]
GamePhase = Literal[
    "motivation",
    "precheck",
    "track_select",
    "preparation",
    "opening",
    "stable",
    "crisis",
    "report",
]
SessionStatus = Literal["active", "finished", "abandoned"]


class GameMetrics(BaseModel):
    cash: int
    debt: int = 0
    dailyNetCashFlow: int = 0
    weeklyNetCashFlow: int = 0
    survivalDays: int = 0
    familyPressure: int = 0
    energy: int = 80
    reputation: int = 50
    repeatRate: float = 0.1
    information: int = 20
    impulse: int = 50


class GameSession(BaseModel):
    id: str
    contentVersion: str
    mode: GameMode
    status: SessionStatus
    phase: GamePhase
    day: int
    week: int
    roleId: str
    businessTrackId: str | None = None
    openingPackageId: str | None = None
    metrics: GameMetrics
    flags: dict[str, bool | int | float | str] = Field(default_factory=dict)


class CreateSessionRequest(BaseModel):
    mode: GameMode = "quick"
    roleId: str
    businessTrackId: str | None = None
    openingPackageId: str | None = None
    initialOverrides: dict[str, int | float | str] = Field(default_factory=dict)


class UpdateSessionRequest(BaseModel):
    phase: GamePhase | None = None
    businessTrackId: str | None = None
    openingPackageId: str | None = None
    flags: dict[str, bool | int | float | str] | None = None


class DecisionCard(BaseModel):
    id: str
    title: str
    phase: GamePhase
    description: str
    cost: dict[str, int | float] = Field(default_factory=dict)
    effect: dict[str, int | float] = Field(default_factory=dict)
    risk: list[str] = Field(default_factory=list)
    processLogs: list[str] = Field(default_factory=list)
    insight: str = ""


class EventCard(BaseModel):
    id: str
    phase: GamePhase
    category: str
    title: str
    content: str
    effect: dict[str, int | float] = Field(default_factory=dict)
    weight: int = 100


class AvailableDecisionsResponse(BaseModel):
    phase: GamePhase
    day: int
    week: int
    event: EventCard | None = None
    decisions: list[DecisionCard]


class SimulateRequest(BaseModel):
    decisionIds: list[str]
    eventId: str | None = None


class LedgerEntry(BaseModel):
    periodType: Literal["day", "week"]
    periodIndex: int
    revenueAmount: int = 0
    materialCost: int = 0
    laborCost: int = 0
    rentCost: int = 0
    platformFee: int = 0
    discountCost: int = 0
    marketingCost: int = 0
    wasteCost: int = 0
    utilityCost: int = 0
    repairCost: int = 0
    operationNetCashFlow: int = 0
    familyCost: int = 0
    loanPayment: int = 0
    familyNetCashFlow: int = 0
    detail: dict[str, int | float | str | list[str]] = Field(default_factory=dict)


class RiskAlert(BaseModel):
    level: Literal["info", "warning", "danger"]
    message: str


class EventResult(BaseModel):
    title: str
    message: str


class SimulateResponse(BaseModel):
    session: GameSession
    eventResult: EventResult
    actionLogs: list[str]
    ledger: LedgerEntry
    riskAlerts: list[RiskAlert]


class Report(BaseModel):
    endingType: str
    title: str
    summary: str
    totalRevenue: int
    totalNetCashFlow: int
    maxLossSources: list[str]
    rightDecisions: list[str]
    dangerousDecisions: list[str]
    suggestion: str
    shareText: str
    reportDetail: dict[str, int | float | str | list[str]]
